x-pack/plugins/cases/server/client/cases/get.ts

```diff
@@ -75,20 +77,53 @@ export const getCaseIDsByAlertID = async (
       Operations.getCaseIDsByAlertID.savedObjectType
     );
 
+    // This will likely only return one comment saved object, the response aggregation will contain
+    // the keys we need to retrieve the cases
     const commentsWithAlert = await caseService.getCaseIdsByAlertId({
       unsecuredSavedObjectsClient,
       alertId: alertID,
       filter,
     });
 
+    // make sure the comments returned have the right owner
     ensureSavedObjectsAreAuthorized(
       commentsWithAlert.saved_objects.map((comment) => ({
         owner: comment.attributes.owner,
         id: comment.id,
       }))
     );
 
-    return CasesService.getCaseIDsFromAlertAggs(commentsWithAlert);
+    const caseIds = CasesService.getCaseIDsFromAlertAggs(commentsWithAlert);
+
+    // if we didn't find any case IDs then let's return early because there's nothing to request
+    if (caseIds.length <= 0) {
+      return [];
+    }
+
+    const casesInfo = await caseService.getCases({
```

Comment:
Now that we fetch the cases I am wondering if there is any benefit with the aggregation?

Reply:
Christos and I talked about this offline and we do still need the aggregation because the alert ID could appear in multiple comment saved objects. So I think we're ok to leave this as it is.